opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -29,12 +42,46 @@ impl<T: Number<T>> Default for ValueMap<T> {
 
 impl<T: Number<T>> ValueMap<T> {
     fn new() -> Self {
+        let buckets = std::iter::repeat_with(|| Mutex::new(None))
+            .take(BUCKET_COUNT)
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap(); // this will never fail as Vec length is fixed
+
         ValueMap {
-            values: Mutex::new(HashMap::new()),
+            buckets: Arc::new(buckets),
             has_no_value_attribute_value: AtomicBool::new(false),
             no_attribute_value: T::new_atomic_tracker(),
+            total_unique_entries: AtomicUsize::new(0),
         }
     }
+
+    // Hash function to determine the bucket
+    fn hash_to_bucket(key: &AttributeSet) -> usize {
+        #[cfg(not(feature = "use_hashbrown"))]
+        let mut hasher = DefaultHasher::new();
+        #[cfg(feature = "use_hashbrown")]
+        let mut hasher = AHasher::default();
+
+        key.hash(&mut hasher);
+        // Use the 8 least significant bits directly, avoiding the modulus operation.
+        hasher.finish() as u8 as usize
+    }
+
+    // Calculate the total length of data points across all buckets.
+    fn total_data_points_count(&self) -> usize {
```

Comment:
is that different from total_unique_entries? 🤔

Reply:
No, this is just a left over, have removed it now.

Reply:
This was a leftover, have removed it now.